### PR TITLE
Jackett All healthcheck and verification

### PR DESCRIPTION
**Pull request by RobinDadswell**
#### Database Migration
NO

#### Description
Jackett all is bad - stopping adding/editing indexers that have this and calls a health check for those that have the all endpoint configured! 

#### Todos
- [ ] Tests
- [x] Translation Keys (./src/NzbDrone.Core/Localization/Core/en.json)
- [x] [Wiki Updates](https://wiki.servarr.com)
- [x] Verification for indexer settings

### CODE REVIEW

**Comment by bakerboy448** [comment]
- [ ] - healthcheck does not work

Existing Indexer 1
- URL `http://127.0.0.1:9117/api/v2.0/indexers/all/results/torznab`
- API `/api`

Existing Indexer 2
- URL `http://localhost:9117/torznab/all/api`
- API `/api`

- [X] - validation does work

New Indexer 3 ([as used by a user in sonarr](https://discord.com/channels/383686866005917708/383688189941907459/919984913330077696))
- URL `http://localhost:9117`
- API `/torznab/all/api`
image


These should also all be added as tests

**Comment by bakerboy448** [comment]
No change from previous comment other than we now validate by API rather than host, which is a regression

healthcheck still does not trigger like it did before

image

**Comment by bakerboy448** [comment]
```
            var jacketAllProviders = _providerFactory.GetAvailableProviders().Where(
                i => i.Definition.ConfigContract.Equals("TorznabSettings"));

                // &&
                // ((i.Definition.Settings as TorznabSettings).BaseUrl.Contains("/torznab/all/api") ||
                // (i.Definition.Settings as TorznabSettings).BaseUrl.Contains("/api/v2.0/indexers/all/results/torznab") ||
                // (i.Definition.Settings as TorznabSettings).ApiPath.Contains("/torznab/all/api") ||
                // (i.Definition.Settings as TorznabSettings).ApiPath.Contains("/api/v2.0/indexers/all/results/torznab")));
```

jacketAllProviders  is empty

**Comment by RobinDadswell** [comment]
> ```
> 
>             var jacketAllProviders = _providerFactory.GetAvailableProviders().Where(
> 
>                 i => i.Definition.ConfigContract.Equals("TorznabSettings"));
> 
> 
> 
>                 // &&
> 
>                 // ((i.Definition.Settings as TorznabSettings).BaseUrl.Contains("/torznab/all/api") ||
> 
>                 // (i.Definition.Settings as TorznabSettings).BaseUrl.Contains("/api/v2.0/indexers/all/results/torznab") ||
> 
>                 // (i.Definition.Settings as TorznabSettings).ApiPath.Contains("/torznab/all/api") ||
> 
>                 // (i.Definition.Settings as TorznabSettings).ApiPath.Contains("/api/v2.0/indexers/all/results/torznab")));
> 
> ```
> 
> 
> 
> jacketAllProviders  is empty

Yes it will be until it hits the foreach loop at which point it will run those lambda expressions

**Comment by bakerboy448** [comment]
I'm saying the healthcheck doesn't work and there is not a single provider found before it checks the other criteria based on commenting those out.  It finds no torznab settings indexers

**Comment by bakerboy448** [comment]
should add some tests otherwise works good

> Indexers using the unsupported Jackett 'all' endpoint: Jackett, Jackett2

- [X] healthcheck good
- [X] validation good

don't forget to [common] this if its not already and it'll need backporting for sonarr

**Comment by RobinDadswell** [comment]
Already on the tests for the health check, validation tests yet to be looked at

File: frontend/src/System/Status/Health/Health.js [hunk 1]
```diff
@@ -27,6 +27,14 @@ function getInternalLink(source) {
           to="/settings/indexers"
         />
       );
+    case 'IndexerJackettAllCheck':
+      return (
```
Comment (Qstick) [inline frontend/src/System/Status/Health/Health.js:31]: Can combine with above case as the link is identical

File: src/NzbDrone.Core/HealthCheck/Checks/IndexerJackettAllCheck.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        }
+
+        public override HealthCheck Check()
+        {
+            var jackettAllProviders = _providerFactory.All().Where(
+                i => i.ConfigContract.Equals("TorznabSettings") &&
+                ((i.Settings as TorznabSettings).BaseUrl.Contains("/torznab/all/api") ||
+                (i.Settings as TorznabSettings).BaseUrl.Contains("/api/v2.0/indexers/all/results/torznab") ||
+                (i.Settings as TorznabSettings).ApiPath.Contains("/torznab/all/api") ||
+                (i.Settings as TorznabSettings).ApiPath.Contains("/api/v2.0/indexers/all/results/torznab")));
```
Comment (Qstick) [inline src/NzbDrone.Core/HealthCheck/Checks/IndexerJackettAllCheck.cs:31]: Do all these need ToLowerInvariant() prior to compare?

**Review by Qstick — CHANGES_REQUESTED** [review]

**Review by Qstick — APPROVED** [review]

**Pull request merged**